### ISSUE TITLE
Logical nullish assignment together with nullish coalescing produces false positive for "use before assignment"

### ISSUE THREAD

**Issue by AlCalzone** [microsoft/TypeScript#41153]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:** logical nullish assignment use before assigned 

**Code**

```ts
let x: boolean;
declare function getX(): boolean;
declare function maybeGetX(): boolean | undefined;

if (Math.random() > 0.5) {
	x = getX();
}

x; // x is used before being assigned (expected)

x ??= maybeGetX() ?? true;

x; // Error: x is used before being assigned (UNEXPECTED)
```
**Expected behavior:**
No error in the last line

**Actual behavior:**
Error: x is used before being assigned

**Playground Link:** [Playground](https://www.typescriptlang.org/play?ts=4.1.0-dev.20201015#code/DYUwLgBAHgXBBGB7RoCGA7A3AWAFABMQBjYVAJxAgDMBXdIsAS0XQgHNwANACgEo4kKEBhwFipCtToNmrALaoAnvBABxLnwHI0rAD4Q6hKo3Qh8ovIyoRuAWVRgAFgDoyGfIjl8IAPggAGZwBWXggAbzwASCgIAF52DV5RAF88PChMCAB6LOgIRgBnAwKzBBAqREkVEzYIVAKCxjZTfBsQKAAHYjAzXjTcGIB+QfiFZTVEiGGIMDIaEAsBzJyIAFEyMkq4GMLi0pUKqpAauoamlpsAVQA5Vc4ABVWAYQAVVYARXiA)

**Related Issues:**

**Comment by MartinJohns** [comment]
Possibly related: #40494

**Comment by Kingwl** [comment]
Yes. It's the same reason of #40494 

You can try [this](https://www.staging-typescript.org/play?ts=4.1.0-pr-40536-3#code/DYUwLgBAHgXBBGB7RoCGA7A3AKACYgGNhUAnECAMwFd0CwBLRdCAc3AA0AKASjiRRAYc+IqXLVaDJhAC2qAJ7wQAcQ48+yNMwA+EGvgr10IXDmz0KETgFlUYABYA6EhlyIZPCAD4IABkcArNwQAN7YAJBQEAC8rGrcOAC+2NhQmBAA9BnQEPQAznp5JgggFIhkJUYsEKh5efQsxrhWIFAADoRgJtwpUQD8fbFyiirxEAMQYCRUIGZpQA)
